pkg/storage/cache.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storage
+
+import (
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+var _ Store = (*Cache)(nil)
+
+type Cache struct {
```

Comment:
In future we can also add some metrics around this. Mainly to see cache hits/misses.

Reply:
yep agreed